pkg/deploy/scaler/scale_test.go

```diff
@@ -0,0 +1,89 @@
+package scaler
+
+import (
+	"testing"
+	"time"
+
+	ktestclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client/testclient"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+
+	"github.com/openshift/origin/pkg/client/testclient"
+	deploytest "github.com/openshift/origin/pkg/deploy/api/test"
+)
+
+func TestScale(t *testing.T) {
+	tests := []struct {
+		testName               string
+		namespace              string
+		name                   string
+		count                  uint
+		preconditions          *kubectl.ScalePrecondition
+		retry, waitForReplicas *kubectl.RetryParams
+		oc                     *testclient.Fake
+		kc                     *ktestclient.Fake
+		expected               []string
+		kexpected              []string
+		expectedErr            error
+	}{
+		{
+			testName:  "simple scale",
+			namespace: "default",
+			name:      "foo",
+			count:     uint(3),
+			oc:        testclient.NewSimpleFake(deploytest.OkDeploymentConfig(1)),
+			kc:        ktestclient.NewSimpleFake(deploytest.MkDeploymentList(1)),
+			expected: []string{
+				"get-deploymentconfig",
+			},
+			kexpected: []string{
+				"get-replicationController",
+				"update-replicationController",
+			},
```

Comment:
My concerns are more about: You've got some interface method you're testing (`Scale`) and the behavior you're defining for the mock dependencies of the scaler are expressed as an uncommented, uncorrelated list of generic API calls (`get`, `update`, etc.)

What line of code the test scaler does the 4th entry in that list (e.g., a `get` amongst a chain of 3 `get` calls) actually correspond to? You have to do like a linear scan of the code every time you want to figure out what the mock operation actually corresponds to. Then apply that to function calls, other dependencies throughout the code, etc. This gets crazy very quickly as the code grows. When you add a new piece of logic to the scaler that interfaces with a mocked dependency, now you have to go back to your linear scanning to figure out how to inject a new out-of-context mock method into that list, compounding the issue every time.

Compare this with a mocking method whereby `Scaler.updateReplicationController` is mocked directly with a function injection. You know immediately how the mock is set up and where it's called from in code. The order of this call in relation to other mocked calls is almost always irrelevant (and when it is relevant, the ordering assertions can be expressed with code that relates other clearly mock injection points in the test code).

Is the order of the mock calls really as significant as these lists imply? I don't yet see how that property justifies the cognitiive overhead of using them for the majority of cases I've seen.